### PR TITLE
Update model references in docstrings

### PR DESCRIPTION
**Pull request by larrybradley**
### Description

This PR updates docstrings in the `modeling.core` module to prevent Sphinx build failures due to unresolved references in external packages that define models which use `astropy.modeling.Model` (via `Fittable2DModel`) as the base class.
This occurred in `photutils`, where I had to ignore these unresolved references to prevent Sphinx warnings, but the references have no links in the docs.  See https://photutils.readthedocs.io/en/latest/api/photutils.psf.MoffatPSF.html#photutils.psf.MoffatPSF as an example.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/modeling/core.py [hunk 1]
```diff
@@ -1431,9 +1431,9 @@ def inverse(self):
     def inverse(self, value):
         if not isinstance(value, (Model, type(None))):
             raise ValueError(
-                "The ``inverse`` attribute may be assigned a `Model` "
-                "instance or `None` (where `None` explicitly forces the "
-                "model to have no inverse."
+                "The ``inverse`` attribute may be assigned a "
+                "`~astropy.modeling.Model` instance or `None` (where "
```
Comment (pllim) [inline astropy/modeling/core.py:1435]: I don't think you have to change it in an error message? In fact I would argue the backticks are also unnecessary.

Comment (larrybradley) [reply astropy/modeling/core.py:1435]: Yeah, I was wondering why a reference was included in an error message.  I'll just remove it.

Comment (larrybradley) [reply astropy/modeling/core.py]: Fixed, thanks!

**Review by pllim — APPROVED** [review]
LGTM thanks. I'll give modeling maintainers a chance to review.

**Comment by pllim** [comment]
Gave them a chance but they didn't, so merging. Thanks!

**Pull request merged**